LunrCore/Vector.cs

```diff
@@ -189,6 +191,15 @@ public void Upsert(int index, double value)
         /// </summary>
         /// <returns>The array of elements.</returns>
         public double[] ToArray()
-            => _elements.Select(el => el.value).ToArray();
+        {
+            var result = new double[_elements.Count];
+
+            for (int i = 0; i < result.Length; i++)
+            {
+                result[i] = _elements[i].value;
+            }
+
+            return result;
+        }
```

Comment:
Is that really better?

Reply:
I'll add a benchmark for this. It eliminates at least one list allocation and possibility more if the concat result  is > 4 elements.

Reply:
Benchmarking shows around a 18% improvement using a custom concat operation with 5 elements, and greater gains as the element count increase in addition to the allocation reductions.


```
BenchmarkDotNet=v0.13.1, OS=debian 11 (container)
Intel Xeon Platinum 8168 CPU 2.70GHz, 1 CPU, 4 logical and 2 physical cores
  [Host] : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT

Toolchain=InProcessEmitToolchain  InvocationCount=1  IterationCount=20  
UnrollFactor=1  WarmupCount=2  

|       Method |     Mean |     Error |    StdDev |
|------------- |---------:|----------:|----------:|
|   LinqConcat | 4.475 μs | 0.1780 μs | 0.2050 μs |
| CustomConcat | 3.700 μs | 0.0937 μs | 0.1079 μs |
```